File: todo_flutter/lib/models/task.dart
class Task {
  final String id;
  final String user;
  final String title;
  final String description;
  final String status;
  final DateTime? dueDate;

  Task({
    required this.id,
    required this.user,
    required this.title,
    required this.description,
    required this.status,
    this.dueDate,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['_id'],
      user: json['user'],
      title: json['title'],
      description: json['description'],
      status: json['status'],
      dueDate: json['dueDate'] != null ? DateTime.tryParse(json['dueDate']) : null,
    );
  }
}

File: todo-backend/models/Task.js
const mongoose = require("mongoose");

const taskSchema = new mongoose.Schema({
  user: {
    type: String, // You can later change this to ObjectId if you add user accounts
    required: true
  },
  title: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    default: ""
  },
  status: {
    type: String,
    enum: ["pending", "in-progress", "done"],
    default: "pending"
  },
  dueDate: {
    type: Date,
    default: null
  },
}, { timestamps: true });

module.exports = mongoose.model("Task", taskSchema);
